mlir/lib/Quantum/Transforms/StaticCustomPatterns.cpp

```diff
@@ -40,7 +40,26 @@ struct LowerStaticCustomOp : public OpConversionPattern<StaticCustomOp> {
                                                                rewriter.getF64FloatAttr(param));
             paramValues.push_back(constant);
         }
-
+        if (op.getGateName() == "MultiRZ") {
+            if (paramValues.size() != 1) {
+                op.emitError() << "MultiRZ gate expects exactly one parameter";
+                return failure();
+            }
```

Comment:
I guess this is another advantage of having explicit quantum gate operations, constraints are easier to express and verify automatically. 

No change needed here.